lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1733,9 +1693,17 @@ def _is_promtail_binary_in_charm(self) -> bool:
         """
         return True if Path(BINARY_PATH).is_file() else False
 
-    def _download_and_push_promtail_to_workload(self, event: RelationChangedEvent) -> None:
+    def _download_and_push_promtail_to_workload(self) -> None:
         """Downloads a Promtail zip file and pushes the binary to the workload."""
-        url = json.loads(event.relation.data[event.app].get("data"))["promtail_binary_zip_url"]
+        # Use the first
+        relations = self._charm.model.relations[self._relation_name]
+        if len(relations) > 1:
+            logger.debug(
+                "Multiple log_proxy relations. Getting Promtail from application {}".format(
+                    relations[0].app.name
+                )
+            )
+        url = relations[0].data[relations[0].app].get("promtail_binary_zip_url")
```

Comment:
Are the multiple [0] here safe? Looks a bit odd but maybe ok in this context

Reply:
I agree it looks funny but yeah it's safe think of it as
```
relation = relations[0]
url = relation.data[relation.app].get("promtail_binary_zip_url")
```

Reply:
Which would probably have been better but not worth adding more commits to this PR.